jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/CallbackCompletionHandler.java

```diff
@@ -0,0 +1,874 @@
+
+        @Override
+        public void onDemandCallbackBlocked(Request request, ThreadInfo demandThreadInfo, ThreadInfo runThreadInfo)
+        {
+            LOG.warn("demand callback blocked for: {}{}demanded by: {}{}possibly blocked: {}",
+                request,
+                System.lineSeparator(),
+                demandThreadInfo,
+                System.lineSeparator(),
+                runThreadInfo
+            );
+        }
+
+        @Override
+        public void onWriteBlocked(Request request, ThreadInfo writeThreadInfo, ThreadInfo writingThreadInfo)
+        {
+            LOG.warn("write blocked for: {}{}write by: {}{}possibly blocked: {}",
+                request,
+                System.lineSeparator(),
+                writeThreadInfo,
+                System.lineSeparator(),
+                writingThreadInfo
+            );
+        }
+
+        @Override
+        public void onWriteCallbackNotCompleted(Request request, Throwable writeFailure, ThreadInfo writeThreadInfo)
+        {
+            LOG.warn("write callback not completed for: {}{}write {} by: {}",
+                request,
+                System.lineSeparator(),
+                writeFailure == null ? "succeeded" : "failed with " + writeFailure,
+                writeThreadInfo
+            );
+        }
+
+        @Override
+        public void onWriteCallbackBlocked(Request request, Throwable writeFailure, ThreadInfo writeThreadInfo, ThreadInfo callbackThreadInfo)
+        {
+            LOG.warn("write callback blocked for: {}{}write {} by: {}{}possibly blocked: {}",
+                request,
+                System.lineSeparator(),
+                writeFailure == null ? "succeeded" : "failed with " + writeFailure,
+                writeThreadInfo,
+                System.lineSeparator(),
+                callbackThreadInfo
+            );
+        }
+    }
+
+    private class StateInfo implements Dumpable
+    {
+        private final Request request;
+        private volatile HandlerCallback handlerCallback;
+        private volatile RequestWrapper.DemandCallback demandCallback;
+        private volatile ResponseWrapper.WriteCallback writeCallback;
+
+        private StateInfo(Request request)
+        {
+            this.request = request;
+        }
+
+        @Override
+        public void dump(Appendable out, String indent) throws IOException
+        {
+            Dumpable demandDumpable = demandCallback;
+            if (demandDumpable == null)
+                demandDumpable = (o, i) -> o.append("demand: %s\n".formatted(demandCallbackTimeout == null ? "not tracked" : "none"));
+
+            Dumpable writeDumpable = writeCallback;
+            if (writeDumpable == null)
+                writeDumpable = (o, i) -> o.append("write: %s\n".formatted(writeCallbackTimeout == null ? "not tracked" : "none"));
+
+            Dumpable.dumpObjects(out, indent, request.toString(), handlerCallback, demandDumpable, writeDumpable);
+        }
+    }
```

Comment:
Avoid capturing lambda:
```suggestion

        private void onComplete(Throwable throwable)
        {
            CallbackCompletionHandler.this.stateInfos.remove(request);
        }
    }
```